server/testdriver/junit4/src/main/java/org/infinispan/server/test/junit4/InfinispanXSiteServerRuleBuilder.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.server.test.junit4;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.infinispan.server.test.core.TestServer;
+
+/**
+ * Builder for {@link InfinispanXSiteServerRule}.
+ *
+ * @author Gustavo Lira &lt;gliraesi@redhat.com&gt;
+ * @since 12.0
+ */
+
+public class InfinispanXSiteServerRuleBuilder {
+
+   private InfinispanXSiteServerRuleBuilder(){}
+
+   private final List<XSite> sites = new ArrayList<>();
+
+   public static InfinispanXSiteServerRuleBuilder sites(XSite... sites) {
+      InfinispanXSiteServerRuleBuilder xSiteServerRuleBuilder = new InfinispanXSiteServerRuleBuilder();
+      Collections.addAll(xSiteServerRuleBuilder.sites, sites);
+      return xSiteServerRuleBuilder;
+   }
+
+   public InfinispanXSiteServerRule build() {
+      Map<String, TestServer> testServers = new HashMap<>();
+      for (XSite xsite: sites) {
+         testServers.put(xsite.getSite(), xsite.build());
```

Comment:
I missed this... it can be a list as well. and checking for duplicates would be nice
(after removing the `XSite` class)
```
   public InfinispanXSiteServerRule build() {
      Set<String> uniqueSiteName = new HashSet<>();
      List<TestServer> sitesTestServers = sites.stream()
            .map(InfinispanServerRuleBuilder::build)
            .map(InfinispanServerRule::getTestServer)
            .peek(testServer -> {
               if (!uniqueSiteName.add(testServer.getSiteName())) {
                  throw new IllegalStateException();
               }
            })
            .collect(Collectors.toList());
      return new InfinispanXSiteServerRule(sitesTestServers);
   }
```

Reply:
@pruivo done